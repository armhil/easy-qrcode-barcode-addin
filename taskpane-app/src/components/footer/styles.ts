import { makeStyles, shorthands } from '@fluentui/react-components';

export const useStyles = makeStyles({
  footer: {
    position: 'absolute',
    width: '-webkit-fill-available',
    bottom: 0,
    ...shorthands.padding('10px'),
    ...shorthands.borderTop('1px', 'solid', 'rgb(204, 204, 204)'),
  },
  footerIcon: {
    float: 'left',
    paddingRight: '8px'
  },
  footerNotification: {
    fontSize: 'small'
  }
});