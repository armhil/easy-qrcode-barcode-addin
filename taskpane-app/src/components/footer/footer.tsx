import React from 'react';
import { Link } from '@fluentui/react-components';
import { LoggingUtils } from 'easy-addins-utils';
import { useStyles } from './styles';
export function GithubFooter() {
  const styles = useStyles();
  const onClick = () => {
    LoggingUtils.Trace('qrbar-github');
    window.open("https://github.com/armhil/easy-qrcode-barcode-addin");
  }

  return (
    <div className={styles.footer}>
      <div className={styles.footerIcon}>
        <img alt="github icon" src="https://github.githubassets.com/images/modules/logos_page/GitHub-Mark.png" width="32px" height="32px"/>
      </div>
      <span className={styles.footerNotification}>Easy QR Code Barcode is open source now! </span>
      <Link as="button" onClick={onClick}>Contribute here</Link>
    </div>
  );
}