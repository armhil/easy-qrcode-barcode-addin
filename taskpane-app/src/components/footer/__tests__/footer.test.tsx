import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { LoggingUtils } from 'easy-addins-utils';
import { GithubFooter } from '../footer';

describe('footer rendering', () => {
  it('should match with snapshot', () => {
    const tree = renderer.create(
      <GithubFooter />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a link as button', () => {
    const dom = render(
      <GithubFooter />
    );

    const githubLink = dom.queryByTestId('contribute-github-button');
    expect(githubLink).not.toBeUndefined();
  });

  it('clicking link should track and open github on production', () => {
    const dom = render(
      <GithubFooter />
    );

    const mockTrackFn = jest.fn();
    LoggingUtils.Trace = mockTrackFn;
    const mockOpenFn: any = jest.fn();
    (window as any).open = mockOpenFn;

    const githubLink = dom.queryByTestId('contribute-github-button');
    githubLink?.click();
    expect(mockTrackFn).toHaveBeenCalledTimes(1);
    expect(mockTrackFn).toHaveBeenCalledWith('qrbar-github');
    expect(mockOpenFn).toHaveBeenCalledWith('https://github.com/armhil/easy-qrcode-barcode-addin');
  });
})