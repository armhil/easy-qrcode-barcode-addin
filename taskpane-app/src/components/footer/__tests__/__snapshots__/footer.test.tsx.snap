// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer rendering should match with snapshot 1`] = `
<div
  className="___1kq0q45_0000000 f1euv43f f1s0s64g f1yab3r1 f1809wu7 f81rol6 f1fow5ox frdkuqy f5ogflp fzkkow9 fxs82t0"
>
  <div
    className="___1nh4tgk_0000000 f6ogs8q f19lj068"
  >
    <img
      alt="github icon"
      height="32px"
      src="https://github.githubassets.com/images/modules/logos_page/GitHub-Mark.png"
      width="32px"
    />
  </div>
  <span
    className="___1opxvxh_0000000 f1bc2y4q"
  >
    Easy QR Code Barcode is open source now! 
  </span>
  <button
    className="fui-Link ___xrvnvs0_qgv3400 f2hkw1w f3rmtva f1ewtqcl fyind8e f1k6fduh f1w7gpdv fk6fouc fkhj508 figsok6 f1hu3pq6 f11qmguv f19f4twv f1tyq0we f1g0x7ka fhxju0i f1qch9an f1cnd47f fqv5qza f1vmzxwi f1o700av f1iuv45f f1cmlufx f9n3di6 f1ids18y f1tx3yz7 f1deo86v f1eh06m1 f1iescvh fhgqx19 f1olyrje f1p93eir f1nev41a f1ern45e f1n71otn f1h8hb77 f1deefiw"
    disabled={false}
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    Contribute here
  </button>
</div>
`;
